torchinductor/codecache.py

```diff
@@ -139,19 +148,24 @@ class CppCodeCache:
     @classmethod
     def load(cls, source_code):
         key, input_path = write(source_code, "cpp", extra=cpp_compile_command("i", "o"))
-        if key not in cls.cache:
-            output_path = input_path[:-3] + "so"
-            if not os.path.exists(output_path):
-                cmd = cpp_compile_command(input=input_path, output=output_path).split(
-                    " "
-                )
-                try:
-                    subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-                except subprocess.CalledProcessError as e:
-                    raise exc.CppCompileError(cmd, e.output)
-
-            cls.cache[key] = cdll.LoadLibrary(output_path)
-            cls.cache[key].key = key
+        lock_dir = os.path.join(cache_dir(), "locks")
+        if not os.path.exists(lock_dir):
+            os.makedirs(lock_dir, exist_ok=True)
```

Comment:
Duplicate logic of creating the lock_dir. How about abstracting a function for this?